hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/index/FlinkHoodieIndexFactory.java

```diff
@@ -49,8 +52,14 @@ public static HoodieIndex createIndex(HoodieFlinkEngineContext context, HoodieWr
         return new FlinkInMemoryStateIndex(context, config);
       case BLOOM:
         return new HoodieBloomIndex(config, ListBasedHoodieBloomIndexHelper.getInstance());
+      case GLOBAL_BLOOM:
+        return new HoodieGlobalBloomIndex(config, ListBasedHoodieBloomIndexHelper.getInstance());
       case SIMPLE:
         return new HoodieSimpleIndex(config, Option.empty());
+      case GLOBAL_SIMPLE:
+        return new HoodieGlobalSimpleIndex(config, Option.empty());
+      case BUCKET:
+        return new HoodieSimpleBucketIndex(config);
```

Comment:
@danny0405 Does Hudi on Flink support Global indexes by design?

Reply:
Flink does not support bloom filter index yet, but it is okey we fix the code now.